langchain/evaluation/run_evaluators/config.py

```diff
@@ -0,0 +1,78 @@
+"""Configuration for run evaluators."""
+from typing import Any, Dict, List, Optional
+
+from langsmith import RunEvaluator
+from pydantic import BaseModel
+
+from langchain.embeddings.base import Embeddings
+from langchain.evaluation.criteria.eval_chain import CRITERIA_TYPE
+from langchain.evaluation.embedding_distance.base import EmbeddingDistance
+from langchain.evaluation.schema import EvaluatorType
+from langchain.evaluation.string_distance.base import StringDistance
+from langchain.schema.language_model import BaseLanguageModel
+from langchain.schema.prompt_template import BasePromptTemplate
+
+
+class EvalConfig(BaseModel):
+    """Configuration for a given run evaluator."""
+
+    evaluator_type: EvaluatorType
+    """The type of evaluator to use."""
+
+    def get_kwargs(self) -> Dict[str, Any]:
+        return self.dict(exclude={"evaluator_type"}, exclude_none=True)
+
+
+class RunEvalConfig(BaseModel):
+    """Configuration for a run evaluation."""
+
+    evaluator_configs: List[EvalConfig]
```

Comment:
Could union type this as =Union[EvalConfig, EvaluatorType] so you could do something like

```
valuation_config = RunEvalConfig(
    evaluator_configs=[
        "qa",
        "context_qa",
        "embedding_distance",
           RunEvalConfig.Criteria(
            criteria={"my-criterion": "Is the answer fewer than 10 words?"}
        ),
    ]
)
```

If the general UX here looks OK, is that UX tradeoff  desired? I think yes.